cluster/create_workflow.go

```diff
@@ -265,3 +293,78 @@ func (a SetupPrivilegesActivity) Execute(ctx context.Context, input SetupPrivile
 
 	return nil
 }
+
+const LabelNodesWithNodepoolNameActivityName = "label-nodes-with-nodepool-name-legacy"
+
+type LabelNodesWithNodepoolNameActivityInput struct {
+	ClusterID uint
+	SecretID  string
+}
+
+type LabelNodesWithNodepoolNameActivity struct {
+	clientFactory cluster.ClientFactory
+	manager       *Manager
+}
+
+func NewLabelNodesWithNodepoolNameActivity(clientFactory cluster.ClientFactory, manager *Manager) LabelNodesWithNodepoolNameActivity {
+	return LabelNodesWithNodepoolNameActivity{
+		clientFactory: clientFactory,
+		manager:       manager,
+	}
+}
+
+type nodeNameLister interface {
+	ListNodeNames() (map[string][]string, error)
+}
+
+func (a LabelNodesWithNodepoolNameActivity) Execute(ctx context.Context, input LabelNodesWithNodepoolNameActivityInput) error {
+	logger := activity.GetLogger(ctx).Sugar().With("clusterId", input.ClusterID)
+
+	cluster, err := a.manager.GetClusterByIDOnly(ctx, input.ClusterID)
+	if err != nil {
+		return err
+	}
+
+	nodeNameLister, ok := cluster.(nodeNameLister)
+	if !ok {
+		logger.Warn("cluster does not expose node lists")
+
+		return nil
+	}
+
+	client, err := a.clientFactory.FromSecret(ctx, input.SecretID)
+	if err != nil {
+		return err
+	}
+
+	nodeNames, err := nodeNameLister.ListNodeNames()
+	if err != nil {
+		return err
+	}
+
+	for poolName, nodes := range nodeNames {
+		logger := logger.With("nodepool", poolName)
+		logger.Debug("labeling nodepool")
+
+		for _, nodeName := range nodes {
+			logger := logger.With("node", nodeName)
+			logger.Debug("labeling node")
+			labels := map[string]string{pkgCommon.LabelKey: poolName}
+
+			tokens := make([]string, 0, len(labels))
+			for k, v := range labels {
+				tokens = append(tokens, "\""+k+"\":\""+v+"\"")
```

Comment:
Are we sure `v` should not be sanitized first?

Reply:
This code is copied without changes.